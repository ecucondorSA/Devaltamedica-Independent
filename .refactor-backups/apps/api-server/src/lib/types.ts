import { z } from 'zod';

// Schema para verificar tokens
export const VerifyTokenSchema = z.object({
  token: z.string().min(1, 'Token is required'),
});

// Schema para registro de usuarios
export const RegisterSchema = z.object({
  email: z.string().email('Invalid email format'),
  password: z.string().min(6, 'Password must be at least 6 characters'),
  name: z.string().min(1, 'Name is required'),
  role: z.enum(['patient', 'doctor', 'admin', 'company']),
});

// Tipos de usuario
export type UserRole = 'patient' | 'doctor' | 'admin' | 'company';

export // Removed local interface - using @altamedica/types
import { User } from '@altamedica/types';