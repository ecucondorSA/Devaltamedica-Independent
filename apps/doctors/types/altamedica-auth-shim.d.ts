declare module '@altamedica/auth' {
  export interface SSOConfig {
    appName: string;
    allowedRoles?: string[];
    loginUrl?: string;
    apiUrl?: string;
    publicPaths?: string[];
    debug?: boolean;
    unauthorizedUrl?: string;
    enableCORS?: boolean;
    allowedOrigins?: string[];
  }
  export function createSSOMiddleware(config: SSOConfig): any;
  export const ssoMiddlewareConfig: { matcher: string[] };
}
