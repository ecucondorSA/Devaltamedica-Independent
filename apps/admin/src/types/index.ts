/**
 * @fileoverview Tipos simples para Admin App - Capa de App
 * @description Estos tipos representan la interfaz de la aplicación admin,
 * simplificados para la UI y consumidos a través de adaptadores
 */

// ==================== TIPOS SIMPLES PARA UI ADMIN ====================

/** Usuario simple para vista administrativa */
export interface SimpleUser {
  id: string;
  name: string;
  email: string;
  role: string;
  status: 'active' | 'inactive' | 'suspended';
  lastActivity: string;
  createdAt: string;
}

/** Estadísticas del dashboard administrativo */
export interface AdminStats {
  totalUsers: number;
  activeUsers: number;
  newUsers24h: number;
  totalAppointments: number;
  activeAppointments: number;
  completedAppointments: number;
  systemHealth: 'healthy' | 'warning' | 'critical';
  errorRate: number;
  averageResponseTime: number;
}

/** Información del sistema para monitoreo */
export interface SystemHealth {
  status: 'up' | 'down' | 'degraded';
  services: ServiceHealth[];
  metrics: SystemMetrics;
  alerts: SystemAlert[];
}

/** Estado de un servicio individual */
export interface ServiceHealth {
  name: string;
  status: 'up' | 'down' | 'degraded';
  port?: number;
  responseTime?: number;
  lastCheck: string;
  uptime?: number;
}

/** Métricas del sistema */
export interface SystemMetrics {
  cpuUsage: number;
  memoryUsage: number;
  diskUsage: number;
  networkLatency: number;
  activeConnections: number;
}

/** Alerta del sistema */
export interface SystemAlert {
  id: string;
  severity: 'low' | 'medium' | 'high' | 'critical';
  message: string;
  timestamp: string;
  resolved: boolean;
  service?: string;
}

/** Log de auditoría simplificado */
export interface AuditLogEntry {
  id: string;
  timestamp: string;
  userId: string;
  userName: string;
  action: string;
  resource: string;
  details: string;
  ipAddress: string;
  userAgent: string;
  severity: 'info' | 'warning' | 'error';
}

/** Configuración del sistema */
export interface SystemSettings {
  maintenanceMode: boolean;
  allowRegistration: boolean;
  enableNotifications: boolean;
  maxSessionDuration: number;
  passwordPolicy: PasswordPolicy;
  hipaaCompliance: HIPAASettings;
  general: {
    siteName: string;
    siteUrl: string;
    timezone: string;
    language: string;
    debugMode: boolean;
  };
  updatedAt: string;
  updatedBy: string;
}

/** Política de contraseñas */
export interface PasswordPolicy {
  minLength: number;
  requireUppercase: boolean;
  requireLowercase: boolean;
  requireNumbers: boolean;
  requireSymbols: boolean;
  expirationDays: number;
}

/** Configuración HIPAA */
export interface HIPAASettings {
  auditLogsRetentionDays: number;
  encryptionEnabled: boolean;
  accessLogsEnabled: boolean;
  dataBackupFrequency: 'daily' | 'weekly' | 'monthly';
}

/** Información de sesión activa */
export interface ActiveSession {
  id: string;
  userId: string;
  userName: string;
  userEmail: string;
  ipAddress: string;
  userAgent: string;
  loginTime: string;
  lastActivity: string;
  location?: string;
  device: string;
}

/** Reporte analítico simple */
export interface AnalyticsReport {
  period: string;
  totalUsers: number;
  activeUsers: number;
  newRegistrations: number;
  totalAppointments: number;
  completedAppointments: number;
  cancelledAppointments: number;
  averageSessionDuration: number;
  topPages: PageAnalytics[];
  userGrowth: GrowthData[];
}

/** Analíticas de página */
export interface PageAnalytics {
  path: string;
  views: number;
  uniqueUsers: number;
  averageTimeOnPage: number;
  bounceRate: number;
}

/** Datos de crecimiento */
export interface GrowthData {
  date: string;
  users: number;
  appointments: number;
  revenue?: number;
}

/** Notificación administrativa */
export interface AdminNotification {
  id: string;
  type: 'info' | 'warning' | 'error' | 'success';
  title: string;
  message: string;
  timestamp: string;
  read: boolean;
  actionUrl?: string;
  actionLabel?: string;
}

/** Filtros para gestión de usuarios */
export interface UserFilters {
  role?: string;
  status?: 'active' | 'inactive' | 'suspended';
  search?: string;
  dateFrom?: string;
  dateTo?: string;
  limit?: number;
  offset?: number;
}

/** Filtros para logs de auditoría */
export interface AuditFilters {
  userId?: string;
  action?: string;
  resource?: string;
  severity?: 'info' | 'warning' | 'error';
  dateFrom?: string;
  dateTo?: string;
  limit?: number;
  offset?: number;
}

/** Respuesta paginada simple */
export interface PaginatedResponse<T> {
  data: T[];
  total: number;
  page: number;
  limit: number;
  hasNext: boolean;
  hasPrevious: boolean;
}

/** Configuración de la app admin */
export interface AdminConfig {
  appName: string;
  version: string;
  environment: 'development' | 'staging' | 'production';
  apiUrl: string;
  enableDebug: boolean;
  features: {
    userManagement: boolean;
    systemMonitoring: boolean;
    auditLogs: boolean;
    analytics: boolean;
    maintenance: boolean;
  };
}

// ==================== ACCIONES Y ESTADOS ====================

/** Estado de carga */
export interface LoadingState {
  loading: boolean;
  error?: string;
  lastUpdated?: string;
}

/** Resultado de una acción administrativa */
export interface AdminActionResult {
  success: boolean;
  message: string;
  data?: unknown;
  errors?: string[];
}

/** Estados de formularios admin */
export interface AdminFormState {
  isSubmitting: boolean;
  isValid: boolean;
  errors: Record<string, string>;
  touched: Record<string, boolean>;
}

// ==================== TIPOS DE EXPORT PARA ADAPTADORES ====================

/** Interface que deben implementar los adaptadores */
export interface AdminAdapter<TComplex, TSimple> {
  toSimple(complex: TComplex): TSimple;
  fromSimple(simple: TSimple): TComplex;
  validateSimple(simple: TSimple): boolean;
}

/** Contexto del admin para providers */
export interface AdminContext {
  user: SimpleUser | null;
  permissions: string[];
  settings: SystemSettings;
  loading: boolean;
  error?: string;
}

// ==================== CONSTANTES Y ENUMS ====================

export const USER_ROLES = {
  SUPER_ADMIN: 'super_admin',
  ADMIN: 'admin',
  DOCTOR: 'doctor',
  PATIENT: 'patient',
  COMPANY: 'company',
  STAFF: 'staff',
} as const;

export const SYSTEM_ALERTS_SEVERITY = {
  LOW: 'low',
  MEDIUM: 'medium',
  HIGH: 'high',
  CRITICAL: 'critical',
} as const;

export const AUDIT_ACTIONS = {
  LOGIN: 'login',
  LOGOUT: 'logout',
  CREATE: 'create',
  UPDATE: 'update',
  DELETE: 'delete',
  VIEW: 'view',
  EXPORT: 'export',
} as const;
