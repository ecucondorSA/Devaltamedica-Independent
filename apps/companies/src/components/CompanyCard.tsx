"use client";

import {
  Building,
  MapPin,
  Users,
  Briefcase,
  Star,
  ExternalLink,
  Heart,
} from "lucide-react";
import Link from "next/link";
import { useState } from "react";

interface Company {
  id: string;
  name: string;
  industry: string;
  description: string;
  location: string;
  size: string;
  rating?: number;
  jobCount?: number;
  logo?: string;
  website?: string;
}

interface CompanyCardProps {
  company: Company;
  animationDelay?: number;
}

export function CompanyCard({ company, animationDelay = 0 }: CompanyCardProps) {
  const [isFavorite, setIsFavorite] = useState(false);
  const [isHovered, setIsHovered] = useState(false);

  const handleFavoriteToggle = (e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    setIsFavorite(!isFavorite);
  };

  return (
    <div
      className="card-default p-6 group cursor-pointer animate-fade-in-up"
      style={{ animationDelay: `${animationDelay}s` }}
      onMouseEnter={() => setIsHovered(true)}
      onMouseLeave={() => setIsHovered(false)}
    >
      {/* Header */}
      <div className="flex items-start justify-between mb-4">
        <div className="flex items-center flex-1">
          <div className="w-16 h-16 bg-gradient-to-br from-sky-500 to-blue-600 rounded-xl mr-4 flex items-center justify-center shadow-lg">
            {company.logo ? (
              <img
                src={company.logo}
                alt={`Logo de ${company.name}`}
                className="w-10 h-10 object-contain"
              />
            ) : (
              <Building className="w-8 h-8 text-white" />
            )}
          </div>
          <div className="flex-1 min-w-0">
            <h3 className="text-xl font-bold text-slate-800 mb-1 truncate group-hover:text-sky-600 transition-colors">
              {company.name}
            </h3>
            <p className="text-sm font-medium text-sky-600 bg-sky-50 px-2 py-1 rounded-full inline-block">
              {company.industry}
            </p>
          </div>
        </div>

        <button
          onClick={handleFavoriteToggle}
          className={`p-2 rounded-full transition-all duration-200 ${
            isFavorite
              ? "text-red-500 bg-red-50 hover:bg-red-100"
              : "text-slate-400 hover:text-red-500 hover:bg-red-50"
          }`}
          aria-label={
            isFavorite ? "Quitar de favoritos" : "Agregar a favoritos"
          }
        >
          <Heart className={`w-5 h-5 ${isFavorite ? "fill-current" : ""}`} />
        </button>
      </div>

      {/* Description */}
      <p className="text-slate-600 mb-4 line-clamp-3 leading-relaxed">
        {company.description}
      </p>

      {/* Company Info */}
      <div className="space-y-3 mb-6">
        <div className="flex items-center text-slate-600">
          <MapPin className="w-4 h-4 mr-3 text-slate-400" />
          <span className="text-sm">{company.location}</span>
        </div>
        <div className="flex items-center text-slate-600">
          <Users className="w-4 h-4 mr-3 text-slate-400" />
          <span className="text-sm">{company.size} empleados</span>
        </div>
        {company.jobCount && (
          <div className="flex items-center text-slate-600">
            <Briefcase className="w-4 h-4 mr-3 text-slate-400" />
            <span className="text-sm">{company.jobCount} ofertas activas</span>
          </div>
        )}
        {company.rating && (
          <div className="flex items-center text-slate-600">
            <Star className="w-4 h-4 mr-3 text-amber-400 fill-current" />
            <span className="text-sm font-medium">{company.rating}/5</span>
          </div>
        )}
      </div>

      {/* Actions */}
      <div className="flex gap-3">
        <Link
          href={`/companies/${company.id}`}
          className="flex-1 btn-primary text-center py-3 text-base"
        >
          Ver Perfil
        </Link>
        <Link
          href={`/companies/${company.id}/jobs`}
          className="flex-1 btn-secondary text-center py-3 text-base"
        >
          Ver Ofertas
        </Link>
      </div>

      {/* Website Link */}
      {company.website && (
        <div className="mt-4 pt-4 border-t border-slate-200">
          <a
            href={company.website}
            target="_blank"
            rel="noopener noreferrer"
            className="inline-flex items-center text-sm text-sky-600 hover:text-sky-800 transition-colors"
            onClick={(e) => e.stopPropagation()}
          >
            <ExternalLink className="w-4 h-4 mr-2" />
            Visitar Sitio Web
          </a>
        </div>
      )}

      {/* Hover Effect */}
      <div
        className={`absolute inset-0 bg-gradient-to-r from-sky-500/5 to-blue-500/5 rounded-2xl transition-opacity duration-300 ${
          isHovered ? "opacity-100" : "opacity-0"
        }`}
      />
    </div>
  );
}
