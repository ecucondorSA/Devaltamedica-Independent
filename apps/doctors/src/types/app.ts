// Re-export appointment types from the main types package
export {
  type Appointment,
  AppointmentStatus,
  AppointmentType,
  type RecurringPattern,
  type NotificationLog,
  type AppointmentOutcome,
  type UrgencyLevel,
  type AppointmentStatus as AppointmentStatusType,
  type AppointmentType as AppointmentTypeType
} from '@altamedica/types/appointment';

// Re-export User type from auth package
export type { User } from '@altamedica/auth';
