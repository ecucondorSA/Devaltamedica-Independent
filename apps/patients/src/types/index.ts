/**
 * Tipos centralizados para Patients App
 * Re-exporta desde @altamedica/types para usar fuente única de verdad
 */

// ✅ TIPOS MÉDICOS CENTRALIZADOS
export type {
  Patient,
  Doctor,
  Appointment,
  MedicalRecord,
  Prescription,
  HealthMetrics,
  VitalSigns,
  LabResult,
  MedicalHistory,
  Medication
} from '@altamedica/types';

// ✅ TIPOS DE AUTENTICACIÓN
export type {
  AuthUser,
  UserRole,
  AuthContext,
  AuthToken
} from '@altamedica/types';

// ✅ TIPOS DE API
export type {
  APIResponse,
  PaginatedResponse
} from '@altamedica/types';

// ✅ Re-export tipos específicos del dashboard (ya correcto)
export * from './dashboard-types';