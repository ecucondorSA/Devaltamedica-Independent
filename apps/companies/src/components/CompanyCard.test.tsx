/**
 * 游빍 ALTAMEDICA COMPANIES - COMPONENT TESTS
 * Tests de integraci칩n para componentes de empresas
 */
import { Company } from '@altamedica/types'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { beforeEach, describe, expect, it, vi } from 'vitest'
import CompanyCard from './CompanyCard'

// Mock del hook de navegaci칩n
const mockPush = vi.fn()
vi.mock('next/navigation', () => ({
  useRouter: () => ({
    push: mockPush,
  }),
}))

// Datos de prueba
const mockCompany: Company = {
  id: '1',
  name: 'Hospital Universitario',
  type: 'hospital',
  address: {
    street: 'Av. Principal 123',
    city: 'Barcelona',
    state: 'Catalu침a',
    zipCode: '08001',
    country: 'Espa침a'
  },
  contactInfo: {
    phone: '+34912345678',
    email: 'info@hospital.com',
    website: 'https://hospital.com'
  },
  businessHours: {
    monday: { open: '08:00', close: '20:00', closed: false },
    tuesday: { open: '08:00', close: '20:00', closed: false },
    wednesday: { open: '08:00', close: '20:00', closed: false },
    thursday: { open: '08:00', close: '20:00', closed: false },
    friday: { open: '08:00', close: '20:00', closed: false },
    saturday: { open: '09:00', close: '14:00', closed: false },
    sunday: { open: '00:00', close: '00:00', closed: true }
  },
  specialties: ['cardiology', 'neurology'],
  services: ['emergency', 'surgery'],
  accreditations: ['joint_commission'],
  certifications: ['iso_9001'],
  staffCount: 150,
  bedsCount: 50,
  foundedYear: 1995,
  description: 'Hospital universitario con excelente atenci칩n m칠dica',
  isVerified: true,
  isActive: true,
  rating: 4.5,
  reviewCount: 150,
  logoUrl: 'https://test.com/logo.png',
  images: ['https://test.com/image1.png'],
  tags: ['quality', 'emergency'],
  metadata: {},
  configuration: {
    allowOnlineBooking: true,
    allowTelehealth: true,
    acceptsInsurance: true,
    emergencyServices: true,
    requiresAppointment: false,
    languages: ['es', 'en', 'ca']
  },
  createdAt: new Date('2023-01-01'),
  updatedAt: new Date('2024-01-01')
}

// Wrapper para React Query
const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: { retry: false },
      mutations: { retry: false },
    },
  })

  const TestWrapper = ({ children }: { children: React.ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      {children}
    </QueryClientProvider>
  )
  TestWrapper.displayName = 'TestWrapper'
  
  return TestWrapper
}

describe('CompanyCard Component', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  const renderComponent = (props = {}) => {
    const Wrapper = createWrapper()
    return render(
      <Wrapper>
        <CompanyCard company={mockCompany} {...props} />
      </Wrapper>
    )
  }

  it('should render company information correctly', () => {
    renderComponent()

    // Verificar informaci칩n b치sica
    expect(screen.getByTestId('company-name')).toHaveTextContent('Hospital Universitario')
    expect(screen.getByTestId('company-type')).toHaveTextContent('Hospital')
    expect(screen.getByTestId('company-address')).toHaveTextContent('Barcelona, Catalu침a')
    expect(screen.getByTestId('company-rating')).toHaveTextContent('4.5')
    expect(screen.getByTestId('company-reviews')).toHaveTextContent('150 rese침as')
  })

  it('should display company specialties', () => {
    renderComponent()

    expect(screen.getByTestId('company-specialties')).toBeInTheDocument()
    expect(screen.getByText('Cardiolog칤a')).toBeInTheDocument()
    expect(screen.getByText('Neurolog칤a')).toBeInTheDocument()
  })

  it('should show verification badge for verified companies', () => {
    renderComponent()

    expect(screen.getByTestId('verified-badge')).toBeInTheDocument()
    expect(screen.getByTestId('verified-badge')).toHaveTextContent('Verificado')
  })

  it('should not show verification badge for unverified companies', () => {
    const unverifiedCompany = { ...mockCompany, isVerified: false }
    renderComponent({ company: unverifiedCompany })

    expect(screen.queryByTestId('verified-badge')).not.toBeInTheDocument()
  })

  it('should display configuration badges', () => {
    renderComponent()

    expect(screen.getByTestId('online-booking-badge')).toBeInTheDocument()
    expect(screen.getByTestId('telehealth-badge')).toBeInTheDocument()
    expect(screen.getByTestId('insurance-badge')).toBeInTheDocument()
    expect(screen.getByTestId('emergency-badge')).toBeInTheDocument()
  })

  it('should handle click to view company details', () => {
    renderComponent()

    const viewDetailsButton = screen.getByTestId('view-company-details')
    fireEvent.click(viewDetailsButton)

    expect(mockPush).toHaveBeenCalledWith('/companies/1')
  })

  it('should handle book appointment action', () => {
    renderComponent()

    const bookButton = screen.getByTestId('book-appointment')
    fireEvent.click(bookButton)

    expect(mockPush).toHaveBeenCalledWith('/companies/1/booking')
  })

  it('should display company logo with fallback', () => {
    renderComponent()

    const logo = screen.getByTestId('company-logo')
    expect(logo).toHaveAttribute('src', 'https://test.com/logo.png')
    expect(logo).toHaveAttribute('alt', 'Logo de Hospital Universitario')
  })

  it('should handle logo error and show fallback', async () => {
    renderComponent()

    const logo = screen.getByTestId('company-logo')
    
    // Simular error de carga de imagen
    fireEvent.error(logo)

    await waitFor(() => {
      expect(screen.getByTestId('company-logo-fallback')).toBeInTheDocument()
    })
  })

  it('should show distance when provided', () => {
    renderComponent({ distance: 2.5 })

    expect(screen.getByTestId('company-distance')).toHaveTextContent('2.5 km')
  })

  it('should not show distance when not provided', () => {
    renderComponent()

    expect(screen.queryByTestId('company-distance')).not.toBeInTheDocument()
  })

  it('should be keyboard accessible', () => {
    renderComponent()

    const card = screen.getByTestId('company-card')
    const viewButton = screen.getByTestId('view-company-details')
    const bookButton = screen.getByTestId('book-appointment')

    // Verificar que los elementos pueden recibir focus
    expect(card).toHaveAttribute('tabIndex', '0')
    expect(viewButton).not.toHaveAttribute('disabled')
    expect(bookButton).not.toHaveAttribute('disabled')

    // Simular navegaci칩n con teclado
    card.focus()
    expect(document.activeElement).toBe(card)

    fireEvent.keyDown(card, { key: 'Enter' })
    expect(mockPush).toHaveBeenCalledWith('/companies/1')
  })

  it('should have proper aria labels for accessibility', () => {
    renderComponent()

    const card = screen.getByTestId('company-card')
    const rating = screen.getByTestId('company-rating')
    const bookButton = screen.getByTestId('book-appointment')

    expect(card).toHaveAttribute('aria-label', expect.stringContaining('Hospital Universitario'))
    expect(rating).toHaveAttribute('aria-label', 'Calificaci칩n: 4.5 de 5 estrellas')
    expect(bookButton).toHaveAttribute('aria-label', 'Reservar cita en Hospital Universitario')
  })

  it('should handle favorite toggle action', async () => {
    const mockToggleFavorite = vi.fn().mockResolvedValue({ isFavorite: true })
    renderComponent({ onToggleFavorite: mockToggleFavorite })

    const favoriteButton = screen.getByTestId('toggle-favorite')
    fireEvent.click(favoriteButton)

    expect(mockToggleFavorite).toHaveBeenCalledWith('1')

    await waitFor(() => {
      expect(screen.getByTestId('favorite-icon-filled')).toBeInTheDocument()
    })
  })

  it('should show loading state while toggling favorite', async () => {
    const mockToggleFavorite = vi.fn().mockImplementation(
      () => new Promise(resolve => setTimeout(resolve, 100))
    )
    renderComponent({ onToggleFavorite: mockToggleFavorite })

    const favoriteButton = screen.getByTestId('toggle-favorite')
    fireEvent.click(favoriteButton)

    expect(screen.getByTestId('favorite-loading')).toBeInTheDocument()
  })

  it('should display business hours indicator', () => {
    renderComponent()

    // Mock current time to be during business hours
    const mockDate = new Date('2024-01-15T10:00:00') // Monday 10 AM
    vi.setSystemTime(mockDate)

    expect(screen.getByTestId('business-hours-indicator')).toBeInTheDocument()
    expect(screen.getByTestId('open-status')).toHaveTextContent('Abierto')
  })

  it('should show closed status outside business hours', () => {
    // Mock current time to be Sunday
    const mockDate = new Date('2024-01-14T10:00:00') // Sunday 10 AM
    vi.setSystemTime(mockDate)

    renderComponent()

    expect(screen.getByTestId('business-hours-indicator')).toBeInTheDocument()
    expect(screen.getByTestId('closed-status')).toHaveTextContent('Cerrado')
  })

  it('should handle responsive design', () => {
    // Mock window.innerWidth
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: 768,
    })

    renderComponent()

    const card = screen.getByTestId('company-card')
    expect(card).toHaveClass('company-card', 'company-card--mobile')
  })

  it('should display emergency services badge when available', () => {
    renderComponent()

    expect(screen.getByTestId('emergency-badge')).toBeInTheDocument()
    expect(screen.getByTestId('emergency-badge')).toHaveTextContent('Emergencias')
  })

  it('should handle company card hover effects', () => {
    renderComponent()

    const card = screen.getByTestId('company-card')
    
    fireEvent.mouseEnter(card)
    expect(card).toHaveClass('company-card--hover')

    fireEvent.mouseLeave(card)
    expect(card).not.toHaveClass('company-card--hover')
  })

  it('should display all supported languages', () => {
    renderComponent()

    const languageList = screen.getByTestId('language-list')
    expect(languageList).toHaveTextContent('Espa침ol')
    expect(languageList).toHaveTextContent('Ingl칠s')
    expect(languageList).toHaveTextContent('Catal치n')
  })

  it('should show staff count when available', () => {
    renderComponent()

    expect(screen.getByTestId('staff-count')).toHaveTextContent('150 profesionales')
  })

  it('should show bed count for hospitals', () => {
    renderComponent()

    expect(screen.getByTestId('beds-count')).toHaveTextContent('50 camas')
  })

  it('should not show bed count for non-hospital types', () => {
    const clinicCompany = { ...mockCompany, type: 'clinic' as const, bedsCount: undefined }
    renderComponent({ company: clinicCompany })

    expect(screen.queryByTestId('beds-count')).not.toBeInTheDocument()
  })

  it('should handle error states gracefully', () => {
    const errorCompany = { ...mockCompany, name: undefined } as any
    
    expect(() => renderComponent({ company: errorCompany })).not.toThrow()
  })

  it('should match snapshot', () => {
    const { container } = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })
})

// Componente CompanyCard simulado para los tests
function CompanyCard({ 
  company, 
  distance, 
  onToggleFavorite 
}: { 
  company: Company
  distance?: number
  onToggleFavorite?: (id: string) => Promise<any>
}) {
  const handleViewDetails = () => {
    mockPush(`/companies/${company.id}`)
  }

  const handleBookAppointment = () => {
    mockPush(`/companies/${company.id}/booking`)
  }

  const handleKeyPress = (e: React.KeyboardEvent) => {
    if (e.key === 'Enter') {
      handleViewDetails()
    }
  }

  const currentTime = new Date()
  const currentDay = currentTime.getDay()
  const days = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday']
  const todayHours = company.businessHours[days[currentDay] as keyof typeof company.businessHours]
  const isOpen = !todayHours.closed && currentTime.getHours() >= 8 && currentTime.getHours() < 20

  return (
    <div 
      data-testid="company-card"
      className="company-card"
      tabIndex={0}
      aria-label={`${company.name} - ${company.type}`}
      onKeyDown={handleKeyPress}
    >
      <div data-testid="company-logo-container">
        <img 
          data-testid="company-logo"
          src={company.logoUrl}
          alt={`Logo de ${company.name}`}
          onError={() => {}}
        />
      </div>

      <div data-testid="company-info">
        <h3 data-testid="company-name">{company.name}</h3>
        <span data-testid="company-type">{company.type === 'hospital' ? 'Hospital' : 'Cl칤nica'}</span>
        
        {company.isVerified && (
          <span data-testid="verified-badge">Verificado</span>
        )}

        <div data-testid="company-address">
          {company.address.city}, {company.address.state}
        </div>

        <div data-testid="company-rating" aria-label={`Calificaci칩n: ${company.rating} de 5 estrellas`}>
          {company.rating}
        </div>
        
        <div data-testid="company-reviews">
          {company.reviewCount} rese침as
        </div>

        {distance && (
          <div data-testid="company-distance">{distance} km</div>
        )}

        <div data-testid="company-specialties">
          {company.specialties.map(specialty => (
            <span key={specialty}>
              {specialty === 'cardiology' ? 'Cardiolog칤a' : 'Neurolog칤a'}
            </span>
          ))}
        </div>

        <div data-testid="configuration-badges">
          {company.configuration.allowOnlineBooking && (
            <span data-testid="online-booking-badge">Reserva Online</span>
          )}
          {company.configuration.allowTelehealth && (
            <span data-testid="telehealth-badge">Telemedicina</span>
          )}
          {company.configuration.acceptsInsurance && (
            <span data-testid="insurance-badge">Acepta Seguros</span>
          )}
          {company.configuration.emergencyServices && (
            <span data-testid="emergency-badge">Emergencias</span>
          )}
        </div>

        <div data-testid="business-hours-indicator">
          {isOpen ? (
            <span data-testid="open-status">Abierto</span>
          ) : (
            <span data-testid="closed-status">Cerrado</span>
          )}
        </div>

        <div data-testid="language-list">
          {company.configuration.languages.map(lang => (
            <span key={lang}>
              {lang === 'es' ? 'Espa침ol' : lang === 'en' ? 'Ingl칠s' : 'Catal치n'}
            </span>
          ))}
        </div>

        {company.staffCount && (
          <div data-testid="staff-count">{company.staffCount} profesionales</div>
        )}

        {company.type === 'hospital' && company.bedsCount && (
          <div data-testid="beds-count">{company.bedsCount} camas</div>
        )}
      </div>

      <div data-testid="company-actions">
        <button 
          data-testid="view-company-details"
          onClick={handleViewDetails}
        >
          Ver Detalles
        </button>
        
        <button 
          data-testid="book-appointment"
          aria-label={`Reservar cita en ${company.name}`}
          onClick={handleBookAppointment}
        >
          Reservar Cita
        </button>

        {onToggleFavorite && (
          <button 
            data-testid="toggle-favorite"
            onClick={() => onToggleFavorite(company.id)}
          >
            <span data-testid="favorite-icon-filled">仇벒잺</span>
          </button>
        )}
      </div>
    </div>
  )
}

export default CompanyCard
