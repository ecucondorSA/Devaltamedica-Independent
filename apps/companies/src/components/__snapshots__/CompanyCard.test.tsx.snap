// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyCard Component > should match snapshot 1`] = `
<div
  aria-label="Hospital Universitario - hospital"
  class="company-card"
  data-testid="company-card"
  tabindex="0"
>
  <div
    data-testid="company-logo-container"
  >
    <img
      alt="Logo de Hospital Universitario"
      data-testid="company-logo"
      src="https://test.com/logo.png"
    />
  </div>
  <div
    data-testid="company-info"
  >
    <h3
      data-testid="company-name"
    >
      Hospital Universitario
    </h3>
    <span
      data-testid="company-type"
    >
      Hospital
    </span>
    <span
      data-testid="verified-badge"
    >
      Verificado
    </span>
    <div
      data-testid="company-address"
    >
      Barcelona
      , 
      Cataluña
    </div>
    <div
      aria-label="Calificación: 4.5 de 5 estrellas"
      data-testid="company-rating"
    >
      4.5
    </div>
    <div
      data-testid="company-reviews"
    >
      150
       reseñas
    </div>
    <div
      data-testid="company-specialties"
    >
      <span>
        Cardiología
      </span>
      <span>
        Neurología
      </span>
    </div>
    <div
      data-testid="configuration-badges"
    >
      <span
        data-testid="online-booking-badge"
      >
        Reserva Online
      </span>
      <span
        data-testid="telehealth-badge"
      >
        Telemedicina
      </span>
      <span
        data-testid="insurance-badge"
      >
        Acepta Seguros
      </span>
      <span
        data-testid="emergency-badge"
      >
        Emergencias
      </span>
    </div>
    <div
      data-testid="business-hours-indicator"
    >
      <span
        data-testid="closed-status"
      >
        Cerrado
      </span>
    </div>
    <div
      data-testid="language-list"
    >
      <span>
        Español
      </span>
      <span>
        Inglés
      </span>
      <span>
        Catalán
      </span>
    </div>
    <div
      data-testid="staff-count"
    >
      150
       profesionales
    </div>
    <div
      data-testid="beds-count"
    >
      50
       camas
    </div>
  </div>
  <div
    data-testid="company-actions"
  >
    <button
      data-testid="view-company-details"
    >
      Ver Detalles
    </button>
    <button
      aria-label="Reservar cita en Hospital Universitario"
      data-testid="book-appointment"
    >
      Reservar Cita
    </button>
  </div>
</div>
`;
